### PR TITLE
feat: request id middleware

### PR DESCRIPTION
**Pull request by craigpastro**
## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: internal/middleware/requestid.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/openfga/openfga/pkg/logger"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const requestIDCtxKey ctxKey = "request-id-context-key"
+
+func NewRequestIDInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
```
Comment (jon-whit) [inline internal/middleware/requestid.go:16]: Mind adding a simple test for this middleware?

File: internal/middleware/requestid.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		err = grpc.SetHeader(ctx, metadata.Pairs("X-Request-Id", requestID))
+		if err != nil {
+			logger.Error("failed to set header", zap.Error(err))
+		}
+
+		return handler(ctx, req)
+	}
+}
+
+func NewStreamingRequestIDInterceptor(logger logger.Logger) grpc.StreamServerInterceptor {
```
Comment (jon-whit) [inline internal/middleware/requestid.go:36]: Mind adding a simple test for this? Just to make sure that the extracted context includes the request id.

File: internal/middleware/requestid.go (hunk 1 above)
Comment (craigpastro) [reply internal/middleware/requestid.go:20]: Now that was interesting. First time writing a middleware test.

File: internal/middleware/requestid.go (hunk 2 above)
Comment (craigpastro) [reply internal/middleware/requestid.go:40]: Yep. Done. thanks!

File: internal/middleware/requestid_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+var pingReq = &pb_testproto.PingRequest{Value: "ping"}
+
+type pingService struct {
+	pb_testproto.TestServiceServer
+	T *testing.T
+}
+
+func (s *pingService) Ping(ctx context.Context, req *pb_testproto.PingRequest) (*pb_testproto.PingResponse, error) {
+	_, ok := ctx.Value(requestIDCtxKey).(string)
```
Comment (jon-whit) [inline internal/middleware/requestid_test.go:23]: We could get double coverage here if you use the context helper methods instead.

```
_, ok := RequestIDFromContext(ctx)
require.True(s.T, ok)
```

Comment (craigpastro) [reply internal/middleware/requestid_test.go]: Done! Thanks!

**Review by jon-whit — APPROVED** [review]

**Pull request merged**